### PR TITLE
feat: more CSAT filters

### PR DESCRIPTION
**Pull request by scmmishra**
This PR contains multiple improvements

1. Refactor the FilterSelector component to have standardized APIs and better abstracted components. (Will be useful once we allow more customized reports)
2. Allow more filters like inbox, team for CSAT

#### Pending Items

- [x] Rating Filters
- [x] Labels Filter
- [x] UI tests
- [x] Backend Tests

https://www.loom.com/share/b36cf32350204f428801bb0c220242a5

### Test vectors considered

- All CSAT filters work without breaking in multiple combinations 
- Conversations report also works perfectly (given FilterSelector is used there too)

### CODE REVIEW

**Comment by scmmishra** [comment]
@nithindavid the functionality works, I'll write the tests and update the PR, meanwhile can you take a look?

**Comment by muhsin-k** [comment]
@scmmishra There is some inconsistency while filtering the reports based on rating.

https://github.com/chatwoot/chatwoot/assets/12408980/897ed217-bc58-4b61-a214-a3c0d4503fc9

**Review by muhsin-k — APPROVED** [review]
I have tested in local and staging environments.

**Comment by scmmishra** [comment]
@nithindavid @pranavrajs one of you PTAL, will merge after that

File: app/javascript/dashboard/routes/dashboard/settings/reports/CsatResponses.vue [hunk 1]
```diff
@@ -35,66 +37,85 @@ export default {
...
         page: this.pageIndex,
-        from: this.from,
-        to: this.to,
-        user_ids: this.userIds,
+        ...this.requestPayload,
+      });
+    },
+    downloadReports() {
+      const type = 'csat';
+      this.$store.dispatch('csat/downloadCSATReports', {
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/reports/CsatResponses.vue:85]: @scmmishra Do we need a try catch here?

**Review by nithindavid — COMMENTED** [review]
Tested on local and Looks good overall. I see couple of API calls without try catch, Shouldn't we need to alert the api call failures?

File: app/javascript/dashboard/routes/dashboard/settings/reports/CsatResponses.vue (hunk 1 above)
Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/settings/reports/CsatResponses.vue:85]: Yeah, makes sense. Making the change

Comment (scmmishra) [reply app/javascript/dashboard/routes/dashboard/settings/reports/CsatResponses.vue]: @nithindavid made the changes

**Comment by pranavrajs** [comment]
@scmmishra I'm getting the following error when deploying. Can you please check?

```
ModuleNotFoundError: Module not found: Error: Can't resolve '../../../../mixins/alertMixin' in '/tmp/build_211ad8b2/app/javascript/dashboard/routes/dashboard/settings/reports'
           at /tmp/build_211ad8b2/node_modules/webpack/lib/Compilation.js:925:10
           at /tmp/build_211ad8b2/node_modules/webpack/lib/NormalModuleFactory.js:401:22
           at /tmp/build_211ad8b2/node_modules/webpack/lib/NormalModuleFactory.js:130:21
           at /tmp/build_211ad8b2/node_modules/webpack/lib/NormalModuleFactory.js:224:22
           at /tmp/build_211ad8b2/node_modules/neo-async/async.js:2830:7
           at /tmp/build_211ad8b2/node_modules/neo-async/async.js:6877:13
           at /tmp/build_211ad8b2/node_modules/webpack/lib/NormalModuleFactory.js:214:25
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/Resolver.js:213:14
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/Resolver.js:285:5
           at eval (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/UnsafeCachePlugin.js:44:7
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/Resolver.js:285:5
           at eval (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/Resolver.js:285:5
           at eval (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/Resolver.js:285:5
           at eval (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/RootPlugin.js:37:38
           at _next43 (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:8:1)
           at eval (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:31:1)
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/Resolver.js:285:5
           at eval (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:67:43
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/Resolver.js:285:5
           at eval (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/Resolver.js:285:5
           at eval (eval at create (/tmp/build_211ad8b2/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/DirectoryExistsPlugin.js:27:15
           at /tmp/build_211ad8b2/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:85:15
           at processTicksAndRejections (node:internal/process/task_queues:78:11)
       resolve '../../../../mixins/alertMixin' in '/tmp/build_211ad8b2/app/javascript/dashboard/routes/dashboard/settings/reports'
         using description file: /tmp/build_211ad8b2/package.json (relative path: ./app/javascript/dashboard/routes/dashboard/settings/reports)
           Field 'browser' doesn't contain a valid alias configuration
           using description file: /tmp/build_211ad8b2/package.json (relative path: ./app/javascript/dashboard/mixins/alertMixin)
             no extension
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin doesn't exist
             .vue
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.vue doesn't exist
             .mjs
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.mjs doesn't exist
             .js
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.js doesn't exist
             .sass
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.sass doesn't exist
             .scss
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.scss doesn't exist
             .css
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.css doesn't exist
             .module.sass
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.module.sass doesn't exist
             .module.scss
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.module.scss doesn't exist
             .module.css
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.module.css doesn't exist
             .png
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.png doesn't exist
             .svg
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.svg doesn't exist
             .gif
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.gif doesn't exist
             .jpeg
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.jpeg doesn't exist
             .jpg
               Field 'browser' doesn't contain a valid alias configuration
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin.jpg doesn't exist
             as directory
               /tmp/build_211ad8b2/app/javascript/dashboard/mixins/alertMixin doesn't exist
       DEPRECATION WARNING: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.
```

**Comment by scmmishra** [comment]
@pranavrajs fixed the issue PTAL

**Review by nithindavid — APPROVED** [review]

**Pull request merged**